### ISSUE TITLE
toolbar button events android

### ISSUE THREAD

**Issue by elliotstokes** [wix/react-native-navigation#1090]
### Issue Description

When adding toolbar buttons to the navigator bar in android the events fail to fire. iOS fire ok.
I'm calling `props.navigator.setOnNavigatorEvent(this._onNavigatorEvent.bind(this))` in my component constructor.  The drawer button on the left fires correctly but the right buttons dont fire any events.

`_onNavigatorEvent` is not called at all.

### Steps to Reproduce / Code Snippets / Screenshots

Add some right buttons on android and check the _onNavigatorEvent.

---
### Environment
* React Native Navigation version: 2.0.0-experimental.268
* React Native version: 0.42
* Platform(s) (iOS, Android, or both?): android
* Device info (Simulator/Device? OS version? Debug/Release?): simulator

**Comment by guyca** [comment]
Hey @elliotstokes 
Where is this line defined: `props.navigator.setOnNavigatorEvent(this._onNavigatorEvent.bind(this))` ? Each screen that's displayed in a `TopTab` should have it's own navigator event function.

**Comment by elliotstokes** [comment]
its defined in the S1 screen.
Below is what we start the app with.
```
Navigation.startSingleScreenApp({
    screen: {
      screen: 'S1',
      title: 'title',
      topTabs: [
        {
          screenId: 'S1',
          title: 'S1'
        },
        {
          screenId: 'S2',
          title: 'S2'
        },
        {
          screenId: 'S2',
          title: 'S3'
        }
      ]
    },
    drawer: {
      left: {
        screen: 'drawer' // unique ID registered with Navigation.registerScreen
      }
    }
  })
```

**Comment by guyca** [comment]
hmm... I wonder if this bit of code is related:
```js
 screen: {
      screen: 'S1',
      title: 'title',
      topTabs: [
        {
          screenId: 'S1',
          title: 'S1'
        },
...
    ]
}
```

Both are the same screen. Could you try using different screen as root? Keep `S1` and `S2` as topTab screens, use `S0` as sort of a "dummy" root screen which is basically only used to declare navigatorStyle

**Comment by elliotstokes** [comment]
Doing the above causes all the buttons to dissapear other than the fab.

**Comment by elliotstokes** [comment]
I'm guessing its also related to this https://github.com/wix/react-native-navigation/issues/940

**Comment by guyca** [comment]
@elliotstokes 
You're right, static navigatorButtons aren't supported in topTabs screens yet. For now, you call `setButtons` in `componentDidMount` of your first screen, and when tab changes.
You can listen to topTab change events in your onNavigatorEvent function of each topTab screen.

```js
onNavigatorEvent(event) {
    if (event.id == 'tabSelected') {
      this.onTabSelected();
    }
}
```

**Comment by elliotstokes** [comment]
Your work around has solved the issue for the time being. Cheers. Much appreciated.

**Comment by shafy** [comment]
I think I have a similar issue: Navigator Event for the current screen (not the modal) is not fired when I open and dismiss a modal. Works fine on iOS, but not on Android. Any more insight regarding this issue yet?
